store/src/java/com/zimbra/cs/service/mail/FileSharedWithMe.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * ***** BEGIN LICENSE BLOCK *****
+ * Zimbra Collaboration Suite Server
+ * Copyright (C) 2022 Synacor, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software Foundation,
+ * version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ * ***** END LICENSE BLOCK *****
+ */
+package com.zimbra.cs.service.mail;
+
+import java.util.Map;
+
+import com.zimbra.common.mailbox.FolderConstants;
+import com.zimbra.common.service.ServiceException;
+import com.zimbra.common.soap.Element;
+import com.zimbra.common.soap.MailConstants;
+import com.zimbra.cs.account.Account;
+import com.zimbra.cs.account.Provisioning;
+import com.zimbra.cs.mailbox.Mailbox;
+import com.zimbra.cs.mailbox.MailboxManager;
+import com.zimbra.cs.mailbox.OperationContext;
+import com.zimbra.soap.ZimbraSoapContext;
+
+public class FileSharedWithMe extends MailDocumentHandler implements DelegatableRequest {
+
+    private static final String REVOKE = "revoke";
+    private static final String EDIT = "edit";
+    private static final String CREATE = "create";
+
+    public Element handle(Element request, Map<String, Object> context) throws ServiceException {
+        try {
+            ZimbraSoapContext zsc = getZimbraSoapContext(context);
+            String ownerAccountId = zsc.getAuthtokenAccountId();
+            String granteeAccountID = zsc.getRequestedAccountId();
+            OperationContext octxt = getOperationContext(zsc, context);
+            Account account = getRequestedAccount(zsc);
+            Mailbox mbox = MailboxManager.getInstance().getMailboxByAccount(account, false);
+
+            String action = request.getAttribute(MailConstants.E_ACTION);
+            String fileName = request.getAttribute(MailConstants.A_CONTENT_FILENAME);
+            int ownerFileId = request.getAttributeInt(MailConstants.A_ITEMID);
+            String fileUUID = request.getAttribute(MailConstants.A_REMOTE_UUID);
+            String fileOwnerName = request.getAttribute(MailConstants.A_OWNER_NAME);
+            String rights = request.getAttribute(MailConstants.A_RIGHTS);
+            String contentType = request.getAttribute(MailConstants.A_CONTENT_TYPE);
+            long size = request.getAttributeLong(MailConstants.A_SIZE);
+            // boolean isDifferentAccountId = !ownerAccountId.equals(granteeAccountID);
+            if (ownerAccountId.equals(granteeAccountID)) {
```

Comment:
move this check to line 43

Reply:
moved the code after getting ownerAccountId and ownerAccountId from zsc , also added logger statement  before throwing exception